internal/lambdalabs/v1/instance.go

```diff
@@ -3,74 +3,219 @@ package v1
 import (
 	"context"
 	"fmt"
+	"strings"
 	"time"
 
+	openapi "github.com/brevdev/cloud/internal/lambdalabs/gen/lambdalabs"
 	v1 "github.com/brevdev/compute/pkg/v1"
 )
 
 // CreateInstance creates a new instance in Lambda Labs
 // Supported via: POST /api/v1/instance-operations/launch
-func (c *LambdaLabsClient) CreateInstance(_ context.Context, attrs v1.CreateInstanceAttrs) (*v1.Instance, error) {
-	// TODO: Implement Lambda Labs instance creation
-	// This would typically involve:
-	// 1. Validating the instance type and location
-	// 2. Creating the instance via Lambda Labs API
-	// 3. Waiting for the instance to be ready
-	// 4. Returning the instance details
+func (c *LambdaLabsClient) CreateInstance(ctx context.Context, attrs v1.CreateInstanceAttrs) (*v1.Instance, error) {
+	keyPairName := attrs.RefID
+	if attrs.KeyPairName != nil {
+		keyPairName = *attrs.KeyPairName
+	}
 
-	return &v1.Instance{
-		Name:         attrs.Name,
-		RefID:        attrs.RefID,
-		CreatedAt:    time.Now(),
-		CloudID:      v1.CloudProviderInstanceID("lambda-instance-id"), // TODO: Get from API response
-		Location:     attrs.Location,
-		SubLocation:  attrs.SubLocation,
-		InstanceType: attrs.InstanceType,
-		ImageID:      attrs.ImageID,
-		DiskSize:     attrs.DiskSize,
-		Status: v1.Status{
-			LifecycleStatus: v1.LifecycleStatusRunning,
-		},
-		Tags: attrs.Tags,
-	}, nil
+	if attrs.PublicKey != "" {
+		request := openapi.AddSSHKeyRequest{
+			Name:      keyPairName,
+			PublicKey: &attrs.PublicKey,
+		}
+
+		_, resp, err := c.client.DefaultAPI.AddSSHKey(c.makeAuthContext(ctx)).AddSSHKeyRequest(request).Execute()
+		if resp != nil {
+			defer func() { _ = resp.Body.Close() }()
+		}
+		if err != nil && !strings.Contains(err.Error(), "name must be unique") {
+			return nil, fmt.Errorf("failed to add SSH key: %w", err)
+		}
+	}
+
+	location := attrs.Location
+	if location == "" {
+		location = "us-west-1"
+	}
+
+	quantity := int32(1)
+	request := openapi.LaunchInstanceRequest{
+		RegionName:       location,
+		InstanceTypeName: attrs.InstanceType,
+		SshKeyNames:      []string{keyPairName},
+		Quantity:         &quantity,
+		FileSystemNames:  []string{},
+	}
+
+	if attrs.Name != "" {
+		request.Name = *openapi.NewNullableString(&attrs.Name)
```

Comment:
look at devplane and use the naming schema there (after cloudcred is implemented and we have the reference id). Notice that we read the name out